### PR TITLE
Set req.ContentLength from headers regardless of body type

### PR DESCRIPTION
**Pull request by majedev**
This is a fix for #822. This makes requests for any generic io.Reader type possible without having to check the underlying Reader implementation. The `headers` map will already have the correct Content-Length value set, so I just parsed it and passed it to req.ContenLength value to match the access signature.

I've also removed util implementation for io.LimitedReader since it's no longer needed - although I still have other concerns for rewinding arbitrary reader types during retries.

### CODE REVIEW

File: storage/util_1.8.go [hunk 1]
```diff
@@ -1,32 +0,0 @@
...
-import (
-	"io"
-	"io/ioutil"
-	"net/http"
-)
-
-func setContentLengthFromLimitedReader(req *http.Request, lr *io.LimitedReader) {
-	req.ContentLength = lr.N
-	snapshot := *lr
-	req.GetBody = func() (io.ReadCloser, error) {
```
Comment (jhendrixMSFT) [inline storage/util_1.8.go]: Removing this means we can no longer rely on it for retrying requests and have to fall back to reading the entire contents of lr into a buffer which seems like a regression (the rest of the change looks good).

Comment (majedev) [reply storage/util_1.8.go]: I don't think this implementation works with retries. There is no rewinding when GetBody is called, so the second retry will fail if the reader was consumed (partially or completely) during the 1st attempt. Here's a demonstration:

https://play.golang.org/p/6gllphBhVK

Comment (jhendrixMSFT) [reply storage/util_1.8.go]: Yep you are right (thanks for the sample BTW it was very helpful).  The copies of LimitReader point to the same underlying reader which is really the thing that needs to be the snapshot.  So the current implementation is broken.  :(

Comment (majedev) [reply storage/util_1.8.go]: My pleasure! I guess some more work need to be done on go-autorest project side for retriable requests, I've mentioned a couple of suggestions here: https://github.com/Azure/go-autorest/issues/157#issuecomment-339113595

 How do you propose to proceed?

Comment (jhendrixMSFT) [reply storage/util_1.8.go]: We are getting ready to release an updated storage SDK which fixes this issue (we use io.ReadSeekers to solve the retry problem).

**Pull request merged**